static/js/login_or_signup.js

```diff
@@ -45,29 +46,40 @@ var checkKeypressIsSpace = function(event) {
 };
 
 $(function() {
-    var $signupForm = $('#signup-form');
-
-    var submitHasBeenClicked = false;
-
     var displayNameField = 'displayName';
     var emailAddressField = 'emailAddress';
     var passwordField = 'password';
 
-    var fields = [
-        displayNameField,
-        emailAddressField,
-        passwordField,
-    ];
-
-    getInputFields($signupForm, fields).forEach(
-        ($field) => {
+    var signupFieldsMetaData = {
+        $form: $('#signup-form'),
+        fields: [
+            displayNameField,
+            emailAddressField,
+            passwordField,
+        ],
+        submitHasBeenClicked: false
+    };
+
+    var loginFieldsMetaData = {
+        $form: $('#login-form'),
+        fields: [
+            emailAddressField,
+            passwordField,
+        ],
+        submitHasBeenClicked: false
+    };
+
+
+    var installFieldValidators = function(fieldsMetaData) {
```

Comment:
how do you feel about `install` vs `create` or any other synonym? `install` feels a bit weird to me

Reply:
install isn't super common in coding, but i feel is extremely accurate for this scenario. Create field handlers would be an excellent name for the anonymous function starting at line 74. But this more than that it doesn't just create them it also installs them. If you want i can call it create and install field validators, but that seems a bit verbose.

Reply:
i'm still not clear on the difference. so what would it mean if i said i created an eventHandler but didn't install it? to me, you didn't create an eventHandler until it's listening for an event.